src/main/java/uk/nhs/digital/nhsconnect/nhais/mapper/AcceptanceCodeMapper.java

```diff
@@ -0,0 +1,32 @@
+package uk.nhs.digital.nhsconnect.nhais.mapper;
+
+import org.hl7.fhir.r4.model.Parameters;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.AcceptanceCode;
+
+import java.util.NoSuchElementException;
+
+public class AcceptanceCodeMapper implements FromFhirToEdifactMapper<AcceptanceCode> {
+    private final static String ACCEPTANCE_CODE = "acceptanceCode";
+
+    public AcceptanceCode map(Parameters parameters) {
+        return AcceptanceCode.builder()
+                .code(getAcceptanceCode(parameters))
+                .build();
+    }
+
+    private String getAcceptanceCode(Parameters parameters) {
+        String inputValue = parameters.getParameter()
+                .stream()
+                .filter(param -> ACCEPTANCE_CODE.equals(param.getName()))
+                .map(Parameters.ParametersParameterComponent::getValue)
+                .map(Object::toString)
+                .findFirst()
+                .orElseThrow();
```

Comment:
Throw what? Will the API response be an OperationOutcome explaining what is missing when this throws?